### ISSUE TITLE
ImageFormatManager.TryFindFormatByFileExtension throws when null or whitespace is bad practice

### ISSUE THREAD

**Issue by bjkippax** [SixLabors/ImageSharp#2385]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

3.0.0

### Other ImageSharp packages and versions

n/a

### Environment (Operating system, version and so on)

Windows 10 & 11, 64bit, build 22000.1574

### .NET Framework version

.NET 7

### Description

The "ImageFormatManager.TryFindFormatByFileExtension" method here throws, which is against the TryX pattern.

```
public bool TryFindFormatByFileExtension(string extension, [NotNullWhen(true)] out IImageFormat? format)
    {
        Guard.NotNullOrWhiteSpace(extension, nameof(extension));

        if (extension[0] == '.')
        {
            extension = extension[1..];
        }

        format = this.imageFormats.FirstOrDefault(x =>
            x.FileExtensions.Contains(extension, StringComparer.OrdinalIgnoreCase));

        return format is not null;
    }
```

What you'd expect is that methods beginning with TryXXXX would never throw and instead simply return false.
Crude example but;

```
public bool TryFindFormatByFileExtension(string extension, [NotNullWhen(true)] out IImageFormat? format)
{
    try
    {
        Guard.NotNullOrWhiteSpace(extension, nameof(extension));

        if (extension[0] == '.')
        {
            extension = extension[1..];
        }

        format = this.imageFormats.FirstOrDefault(x =>
            x.FileExtensions.Contains(extension, StringComparer.OrdinalIgnoreCase));
    catch (Exception)
    {
        // Do nothing
    }

    return format is not null;
}
```

It should be the responsibility of the consuming application to ensure that it has thoroughly tested its implementation of this method.

### Steps to Reproduce

```
// this
ImageFormatManager.TryFindFormatByFileExtension("", out ImageFormat? foo);

// or this
ImageFormatManager.TryFindFormatByFileExtension(null, out ImageFormat? foo);
```

**Comment by JimBobSquarePants** [comment]
Yep. That guard should be removed, and null handled by assigning format to null and returning false. I wouldn't use a try/catch there.

Feel free to open a PR if you have the time.

**Comment by bjkippax** [comment]
@JimBobSquarePants I appreciate your swift response James. The try-catch was a crude example, I would never had intended to suggest that as a real-world fix/suggestion, just to illustrate my point.

One of my colleagues will be submitting a PR for this.

Best,
Ben

**Comment by JimBobSquarePants** [comment]
Legend! Thanks

**Comment by Ollie-Ave** [comment]
Hi, @bjkippax 's colleague here, I have created a PR for this change :). Here's the link:

https://github.com/SixLabors/ImageSharp/pull/2386

**Comment by JimBobSquarePants** [comment]
Fixed via #2386
